### ISSUE TITLE
[CLOSED] multiline edits

### ISSUE THREAD

**Issue by lastland** [plclub/hs-to-coq#32]
<a href="https://github.com/sweirich"><img src="https://avatars1.githubusercontent.com/u/2103670?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [sweirich](https://github.com/sweirich)**
_Monday Oct 16, 2017 at 15:25 GMT_
_Originally opened as https://github.com/antalsz/hs-to-coq/issues/32_

----

Making each edit take exactly one line is harsh for redefine / add edits.   

Can we add a multiline-block to the parser so that instead of 
```
redefine Local Definition instance_GHC_Base_Eq__Boxity_op_zsze__ : Boxity -> (Boxity -> bool) := fun arg_319__ arg_320__ =>  match arg_319__ , arg_320__ with | a , b => negb (instance_GHC_Base_Eq__Boxity_op_zeze__ a b)  end.  
```

we can write 

```
redefine {{ Local Definition instance_GHC_Base_Eq__Boxity_op_zsze__ : Boxity -> (Boxity -> bool) := 
    fun arg_319__ arg_320__ =>  
        match arg_319__ , arg_320__ with 
            | a , b => negb (instance_GHC_Base_Eq__Boxity_op_zeze__ a b)  
        end. 
}}
```

**Comment by lastland** [comment]
<a href="https://github.com/nomeata"><img src="https://avatars2.githubusercontent.com/u/148037?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [nomeata](https://github.com/nomeata)**
_Monday Oct 16, 2017 at 22:56 GMT_

----

Or we could just require a `.` at the end of every edit, instead of looking for newlines. Might be more convenient for the Coq-experience user.

**Comment by lastland** [comment]
<a href="https://github.com/antalsz"><img src="https://avatars2.githubusercontent.com/u/4643974?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [antalsz](https://github.com/antalsz)**
_Tuesday Oct 17, 2017 at 04:29 GMT_

----

I think the happy medium there is to treat newlines as whitespace *iff* we're in a Coq definition, and require a period there.  (I really want to preserve the newlineless syntax for simple things like `skip` and `rename`.)

**Comment by lastland** [comment]
<a href="https://github.com/antalsz"><img src="https://avatars2.githubusercontent.com/u/4643974?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [antalsz](https://github.com/antalsz)**
_Tuesday Oct 17, 2017 at 22:00 GMT_

----

Fixed by bfb586488fa529b829b8d16659ef1c4f52b25c6e – all Coq definitions in the edits file can have embedded newlines and must end with a `.`.
